pep-0590.rst

```diff
@@ -308,7 +308,7 @@ References
 .. [2] tp_call/PyObject_Call calling convention
    https://docs.python.org/3/c-api/typeobj.html#c.PyTypeObject.tp_call
 .. [3] Using PY_VECTORCALL_ARGUMENTS_OFFSET in callee
-   https://github.com/markshannon/cpython/blob/vectorcall-minimal/Objects/classobject.c#L53
+   https://github.com/markshannon/cpython/blob/815cc1a30d85cdf2e3d77d21224db7055a1f07cb/Objects/classobject.c#L53
```

Comment:
Yes, at least pretty forevermore. GitHub describes them as permanent links and permalinks:

https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files

A big downside of the Wayback Machine is it's _much slower_ to load than GitHub. 

And there doesn't appear to be any Wayback Machine link for that page, let alone for when the PEP was written.

Reply:
Thanks for the link, sounds reasonable. (I'd probably prefer that authors used tags or similar going forwards, but for "fixing" previous this may be the least worst solution).

A

Reply:
> I'd probably prefer that authors used tags or similar going forwards, but for "fixing" previous this may be the least worst solution.

Tags are mutable and can change or be deleted, and require the user to either find a suitable tag (if one exists), or (if they control the repo) create one, which is far more effort than simply pressing the `y` key. That's why always using proper permalinks is the standard, best practice approach for links that are, well, permanent. So long as authors are using inline links correctly (which we will make sure they will, going forward) what is the downside to explicitly specifying a commit hash?